React Redux Group Project Submission PR

fickryiman opened the pull request:
- [x] Completed all required tasks
- [x] Each team member has contributed to their tasks
- [x] Managed and work through the Kanban board

fickryiman committed:
create react app with npx create-react-app

fickryiman committed:
Merge pull request #1 from fickryiman/day-0

config and setup project

fickryiman committed:
config and setup linters

fickryiman committed:
update eof

fickryiman committed:
Merge pull request #2 from fickryiman/day-0

config and setup linters

fickryiman committed:
add readme description details

fickryiman committed:
Merge pull request #3 from fickryiman/day-0

add readme description details

fickryiman committed:
update readme file

fickryiman committed:
update readme file

fickryiman committed:
Merge pull request #4 from fickryiman/day-0

update readme file

ehmaddd committed:
Merge pull request #51 from fickryiman/development

Initial settings : Kanban board

ehmaddd committed:
Merge pull request #52 from fickryiman/day-0

Merge pull request #51 from fickryiman/development

ehmaddd committed:
Add only required files

ehmaddd committed:
Merge pull request #53 from fickryiman/setup-setting

Add only required files

ehmaddd committed:
Add Empty My Profile Setup

ehmaddd committed:
Add basic setup files

fickryiman committed:
Merge pull request #54 from fickryiman/My-Profile-Setup

Add Empty My Profile Setup

ehmaddd committed:
Merge pull request #55 from fickryiman/basic-mission-setup

Basic mission setup

fickryiman committed:
create website header

- Download the free image (https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- create empty navigation
- add image for readme file

fickryiman committed:
remove width from sidebar link

ehmaddd committed:
Add fetch data files

ehmaddd committed:
Add mission list display files

ehmaddd committed:
Add Mission Joining Action Updates

ehmaddd committed:
Make Linter Correction

ehmaddd committed:
Update .eslintrc.json

ehmaddd committed:
Make necessary changes

ehmaddd committed:
Make changes

ehmaddd committed:
Add changes

ehmaddd committed:
Add changes

fickryiman committed:
adjust navbar size and position

fickryiman committed:
eslint fixes

ehmaddd committed:
Add updated files

fickryiman committed:
Merge pull request #56 from fickryiman/create-website-header-with-empty-nav

create website header

fickryiman committed:
update images on readme file

fickryiman committed:
Merge pull request #60 from fickryiman/create-website-header-with-empty-nav

update images on readme file

fickryiman committed:
add the basic structure for rockets - setup

fickryiman committed:
Merge pull request #61 from fickryiman/create-basic-structure-for-Rockets

add the basic structure for rockets - setup

ehmaddd committed:
Add mission leaving action code

ehmaddd committed:
Merge pull request #57 from fickryiman/fetch-mission-data

Add fetch data files

ehmaddd committed:
Merge pull request #58 from fickryiman/mission-list-display

Mission list display

ehmaddd committed:
Merge pull request #62 from fickryiman/mission-leaving-action

Mission leaving action

ehmaddd committed:
Add switch badge files

ehmaddd committed:
Merge branch 'development' of github.com:fickryiman/react-redux-group-project into switch-badges

ehmaddd committed:
Add updated files

ehmaddd committed:
Add display mission in profile code

fickryiman committed:
Fetch data from the Rockets endpoint

- add redux rocket slice
- add rockets component

fickryiman committed:
remove boilerplate code

fickryiman committed:
add dispatch getRockets

ehmaddd committed:
Merge pull request #63 from fickryiman/switch-badges

Add switch badge files

ehmaddd committed:
Merge pull request #64 from fickryiman/display-mission-profile

Display mission profile

ehmaddd committed:
Add changes

ehmaddd committed:
Merge branch 'fetch-rockets' of github.com:fickryiman/react-redux-group-project into development

fickryiman committed:
Merge pull request #65 from fickryiman/fetch-rockets

Fetch data from the Rockets endpoint

fickryiman committed:
Render a list of rockets

- add style
- render a list of rockets from api endpoint
- choose image from first flickr-images array

fickryiman committed:
fixes linters

fickryiman committed:
Merge pull request #66 from fickryiman/display-rockets-lists-render

Render a list of rockets

fickryiman committed:
In this PR, the following features are applied :

- On user click "reserve rocket" button, action applied to update the store.
- Got the ID of the selected rocket and updated the state
- Didn't mutate the state. Instead, returned a new state object with all rockets, and the selected rocket has extra key reserved with its value set to true.
- Used map() function to set the value of the new state.
- Place all rocket logic in the reducer folder.
- In the React view file, only the action is dispatched with the correct rocket ID as an argument.

fickryiman committed:
Following features are applied :

- Followed the same logic as with the "reserve rocket" - but set the reserved key to false.
- Dispatched the actions upon click on the corresponding buttons.

fickryiman committed:
add badges after clicked reserve rocket button

Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design)

fickryiman committed:
display reserved rockets on myprofile page

Render a list of all reserved rockets (use filter()) on the "My profile" page.

fickryiman committed:
remove old code

fickryiman committed:
Merge pull request #67 from fickryiman/rocket-booking-actions

Rocket booking actions

fickryiman committed:
Revert "Rocket booking actions"

fickryiman committed:
Merge pull request #74 from fickryiman/revert-67-rocket-booking-actions

Revert "Rocket booking actions"

fickryiman committed:
linter fixes

fickryiman committed:
Merge branch 'development' of https://github.com/fickryiman/react-redux-group-project into display-reserved-rockets-in-my-profile

fickryiman committed:
fixes eslint

fickryiman committed:
Merge pull request #75 from fickryiman/display-reserved-rockets-in-my-profile

Display reserved rockets in my profile

fickryiman committed:
update live demo url on readme file

fickryiman committed:
Merge pull request #79 from fickryiman/finalized-project

update live demo url on readme file

fickryiman committed:
Finalized Project

Finished all the requirements from kanban board

fickryiman committed:
Merge pull request #80 from fickryiman/finalized-project

Finalized Project

Olamarx reviewed (changes requested):
Hi Team, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!
<img src="https://media4.giphy.com/media/hXz6GN0ZilLwhgLo49/giphy.gif"/>

## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Redux usage ✔️
- GitHub flow ✔️

## Required Changes ♻️
_Check the comments under the review._
- [ ] You have probably forgotten to write tests, it is also required to move on.
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

fickryiman committed:
add tests folder

- change component folder to components

fickryiman committed:
linter fixes

fickryiman committed:
linter fixes

fickryiman committed:
Set up testing libraries(React Testing Library and Jest)

- add test for rocket slice
- add test for mission slice
- add snapshot test for rockets ui
- add snapshot test for missions ui
- add snapshot test for myprofile ui

fickryiman committed:
eslint linter fixes

fickryiman committed:
fixes typo

fickryiman committed:
Merge pull request #82 from fickryiman/testing-library-and-jest

Set up testing libraries(React Testing Library and Jest)

ichala reviewed (approved):
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
![68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966](https://user-images.githubusercontent.com/85113567/184604589-430cf4bc-1d2e-4603-976a-7a48a899ea35.gif)

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
